packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -208,17 +208,23 @@ export class PureDataObjectFactory<
 	 * @param context - The context being used to create the runtime
 	 * (the created object will have its own new context created as well)
 	 * @param initialState - The initial state to provide to the created data store.
+	 * @param groupId - group to which this data stores belongs to. The datastores with groupId might not be loaded
+	 * immediately. This groupId is also known at service side and can be used to fetch snapshot contents like
+	 * snapshot tree, blobs using this id from the storage. Snapshot for a datastore with no groupId will definitely
+	 * be included in the snapshot on load.
```

Comment:
I'd love to see a link from here to a readme.md that talks in more details about this feature. It's fine if it's empty for starters, but we need to put really great documentation on how that feature works